src/main/java/duke/main/Utils.java

```diff
@@ -0,0 +1,192 @@
+package duke.main;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import duke.exceptions.*;
+import duke.items.*;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Utils {
+    private static String[] VALID_COMMANDS = {"bye", "done", "list", "todo", "event", "deadline", "delete"};
+
+    public static void printLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void welcomeMessage(){
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm duke.main.Duke\nWhat can I do for you?");
+    }
+
+    public static int validateCommand(String line){
+        if ((Arrays.asList(VALID_COMMANDS).contains(line))) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+    public static int validateParameter(String line) {
+        int indexOfSlash = -1;
+        if (line.split(" ")[0].equals("deadline")) {
+            indexOfSlash = line.indexOf("/by");
+        } else if (line.split(" ")[0].equals("event")) {
+            indexOfSlash = line.indexOf("/at");
+        }
+        return indexOfSlash;
+    }
+    public static void createFile(){
+        try {
+            File myObj = new File("list.txt");
+            if (myObj.createNewFile()) {
+                System.out.println("File created: " + myObj.getName());
+            } else {
+                System.out.println("File already exists.");
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+        }
+    }
+    public static void writeToFile() throws IOException {
+        createFile();
+        FileWriter fw = new FileWriter("list.txt");
+        for (int i=0; i<Task.getNumOfTasks(); i++) {
+            ArrayList<Task> buffer = Task.getList();
+            switch (buffer.get(i).getType()) {
+
+            case ("T"):
+                fw.write(buffer.get(i).getType() + "\t" + buffer.get(i).isDone() + "\t" + buffer.get(i).getDescription() + "\n");
```

Comment:
maybe you should consider improve your code abstraction here. Please check SLAP hard for example. You have fw.write repeated for each case statement.